src/braket/devices/local_simulator.py

```diff
@@ -326,3 +322,37 @@ def _(
             )
         results = simulator.run(program, shots, *args, **kwargs)
         return AnalogHamiltonianSimulationQuantumTaskResult.from_object(results)
+
+    def _prepare_batch_tasks_and_inputs(
```

Comment:
Move this part out of run_batch because flake8 detects this logic as "too complex"

Reply:
okay, thanks, I think flake8 flags is better too. This PR only adds two lines to this function. Moving code shows too many diff that is distracting.

Reply:
Added the flag to remove the need of this private method. This reduces the diff that is not relevant to this PR.